fix: activate community channel link

igor-sirotin opened the pull request:
Fixes https://github.com/status-im/status-desktop/issues/12539

### What does the PR do

1. Improve the algorithm of switching to channel by using the known community id
2. Fix the used `chatID` for chat lookup, as it should be a concatenation of community id and channel id
3. Fix the order of processing channel/community in `activateStatusDeepLink`. Channel should be processed earlier, because communityId is also set
4. Fix json parsing of `channelUuid` field 
5. Refactor switching to community/channel to use `switchTo` function
6. Works for non-spectated/joined community too!

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/25482501/7d5d887c-2664-4ae5-ad95-d042f8f457b8

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 588db3a3 | [#1](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-12653/1/) | 2023-11-07 22:28:26 | ~5 min | `tests/nim` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-12653/1//consoleText) |

MishkaRogachev commented on src/app/modules/main/module.nim:
```diff
@@ -1308,21 +1308,23 @@ method onStatusUrlRequested*[T](self: Module[T], action: StatusUrlAction, commun
-        found = true
-        break
-    if not found:
+    let chatId = communityId & channelId
```
I remember running into that too

MishkaRogachev commented on src/app/modules/main/module.nim:
```diff
@@ -1308,21 +1308,23 @@ method onStatusUrlRequested*[T](self: Module[T], action: StatusUrlAction, commun
+      return
+
+    let cModule = self.channelGroupModules[communityId]
+    let item = self.view.model().getItemById(communityId)
```
can we add a check here is community presented in the model

igor-sirotin replied:
hmmm, we have a check above for `if not self.channelGroupModules.hasKey(communityId)`, I hoped for that one. But probably you're right, we should check the `item.isEmpty` as well.

igor-sirotin replied:
refactored this part to use `switchTo`

borismelnik reviewed (approved):
LGTM :)

igor-sirotin commented:
Just noted that it doesn't work for not yet spectated/joined communities. Fixing.

igor-sirotin commented:
@borismelnik @MishkaRogachev I've updated the PR, please check it out again 🙂 

Now switching to a channel of non-spectated/joined community works as well.
And I refactored things to reuse existing `switchTo` function.

MishkaRogachev reviewed (approved):
LGTM!
